mitwindfarm/CurledWake.py

```diff
@@ -230,35 +231,28 @@ def stamp_ic(
             az=az,
         )
 
-        if self.ic_method == "fx":
-            # NOTE: DO NOT USE
-            thrust = -rotor.Ct * 0.5 * np.pi / 4
-            self.extra_fx += (
-                shape * thrust / (np.sum(shape) * self.dy * self.dz * self.dx)
-            )
-            warn(
-                "`fx` is not a reliable method for stamping initial conditions. Use `du` instead."
-            )
-        else:
-            # stamp the rotor solution into the wind field
-            # TODO: check du is negative?
-            delta_u = rotor.u4 - rotor.REWS  # delta_u, adjusted by REWS
-            self.du[-1, ...] += shape * delta_u
+        # stamp the rotor solution into the wind field
+        # TODO: check du is negative?
+        delta_u = rotor.u4 - rotor.REWS  # delta_u, adjusted by REWS
+        self.du[-1, ...] += shape * delta_u
 
         # dv, dw initial conditions:
-        if rotor.yaw == 0:
+        if (rotor.yaw == 0) and (rotor.tilt == 0):
             return  # no additional dv, dw to stamp in for this turbine
 
         # TODO: Put this in a separate module
-        # self.N_vortex = 10  # make this a parameter
-        # self.vortex_sigma = 0.2  # sigma/D, for de-singularization
 
         # r-axis: clip edges to prevent singularities
-        r_i = np.linspace(-(D - self.dz) / 2, (D - self.dz) / 2, self.N_vortex)
-        # NOTE: rotor.Ct differs from Shapiro et al. (2018) definition - includes cos^2(yaw)
-        Gamma_0 = 0.5 * D * rotor.REWS * rotor.Ct * np.sin(rotor.yaw)
+        d_yx = np.maximum(self.dy, self.dz)
+        # along z-axis in yaw-only frame (also the radial distance of each point from center in any frame)
+        r_i = np.linspace(-(D - d_yx) / 2, (D - d_yx) / 2, self.N_vortex)
+        # rotate points into yaw-and-tilt frame
+        eff_yaw = calc_eff_yaw(rotor.yaw, rotor.tilt)
+        _, y_i, z_i = eff_yaw_inv_rotation(np.zeros_like(r_i), np.zeros_like(r_i), r_i, eff_yaw, rotor.yaw, rotor.tilt)
+        # NOTE: rotor.Ct differs from Shapiro et al. (2018) definition - includes cos^2(eff_yaw)
+        Gamma_0 = 0.5 * D * rotor.REWS * rotor.Ct * np.sin(eff_yaw)
         Gamma_i = (
-            Gamma_0 * 4 * r_i / (self.N_vortex * D**2 * np.sqrt(1 - (2 * r_i / D) ** 2))
+            Gamma_0 * 4 * r_i / (self.N_vortex * D * np.sqrt(1 - (2 * r_i / D) ** 2))
```

Comment:
![IMG_8CE22A855DEF-1](https://github.com/user-attachments/assets/9b7d3a00-87f4-41d2-939b-426f166b51fa)

I am pretty sure that this should just be over `D` due to a cancellation with the Reimann sum. This also makes the units work out since gamma_i should be the same units as gamma_0.